src/Development/Rattle/Server.hs

```diff
@@ -132,7 +132,7 @@ withRattle options@RattleOptions{..} act = withUI rattleFancyUI (return "Running
             (act r <* saveSpeculate state) `finally` writeVar state (Left Finished)
     attempt1 `catch` \(h :: Hazard) -> do
         b <- readIORef speculated
-        if not $ recoverableHazard h then throwIO h else do
+        if not (recoverableHazard h || restartableHazard h) then throwIO h else do
```

Comment:
Why not add `hazardRecoverable :: Hazard -> Recoverable` then do `hazardRecoverable h == Recoverable` as this test.

Reply:
Do you mean hazardRecoverable h == NonRecoverable?  If so, then I agree yes that would maybe be clearer.

Reply:
It was just intended to be a refactoring, not a code change

Reply:
Ok I understand your intention now seeing your other comment.  Then yes that does seem like a nice change.